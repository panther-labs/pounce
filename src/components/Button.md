Button pizza example:

```jsx
<Button pizza="🍕" />
```

Feel free to write anything here btw!
