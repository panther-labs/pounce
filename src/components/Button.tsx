import * as React from 'react';
import { css } from 'styled-components';
import BaseButton, { BaseButtonProps } from 'components/BaseButton';
import { convertHexToRgba } from 'utils/helpers';

export interface ButtonProps extends BaseButtonProps {
  /** The size of the button */
  size: 'small' | 'large';

  /** The variant of the button that decides the colors */
  variant: 'primary' | 'default' | 'secondary';
}

/**
 * Extends <a href="/#/Box">Box</a>
 *
 * The core re-usable button that you will use in the app.
 */
const Button: React.FC<ButtonProps> = ({ size, variant, children, css: userCssProp, ...rest }) => {
  const sizeProps = (() => {
    switch (size) {
      case 'small':
        return { px: 4, py: 2, fontSize: 2 };
      case 'large':
      default:
        return { px: 5, py: 4, fontSize: 3 };
    }
  })();

  const variantProps = (() => {
    switch (variant) {
      case 'secondary':
        return {
          color: 'grey400',
          bg: 'grey50',
          css: css`
            text-transform: uppercase;

            &:hover {
              background-color: ${({ theme }) => theme.colors.grey100};
            }

            &:active {
              background-color: ${({ theme }) => theme.colors.grey200};
            }
          `,
        };
      case 'default':
        return {
          color: 'grey400',
          bg: 'white',
          boxShadow: 'dark150',
          css: css`
            text-transform: uppercase;

            &:hover {
              box-shadow: ${({ theme }) => theme.shadows.dark200};
            }

            &:active {
              box-shadow: ${({ theme }) => theme.shadows.none};
              background-color: ${({ theme }) => theme.colors.grey100};
            }
          `,
        };
      case 'primary':
      default:
        return {
          color: 'white',
          bg: 'primary300',
          boxShadow: 'dark150',
          css: css`
            text-transform: uppercase;

            &:hover {
              box-shadow: ${({ theme }) => theme.shadows.dark200};
              background-color: ${({ theme }) => convertHexToRgba(theme.colors.primary300, 0.9)};
            }
            
            &:active {
              box-shadow: ${({ theme }) => theme.shadows.none};
              background-color: ${({ theme }) => theme.colors.primary300};
            },
          `,
        };
    }
  })();

  return (
    <BaseButton
      fontWeight="bold"
      borderRadius="large"
      css={`
        ${userCssProp}
        ${variantProps.css}
      `}
      {...sizeProps}
      {...variantProps}
      {...rest}
    >
      {children}
    </BaseButton>
  );
};

export default Button;
