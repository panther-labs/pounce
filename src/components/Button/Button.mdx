A button can be primary:

```typescript jsx
<Button size="large" variant="primary">
  Hey there
</Button>
```

A button can be default:

```typescript jsx
<Button size="large" variant="default">
  Hey there
</Button>
```

A button can be small:

```typescript jsx
<Button variant="secondary" size="small">
  Hey there
</Button>
```

A button can contain an icon:

```typescript jsx
<Button variant="primary" size="small" is="div">
  <Flex alignItems="center">
    <Icon type="arrow-back" size="small" mr={1} />
    <Text size="medium">Take me back</Text>
  </Flex>
</Button>
```
