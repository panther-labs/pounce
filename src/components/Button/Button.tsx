import * as React from 'react';
import AbstractButton from '../AbstractButton';
import { BoxProps, NativeAttributes } from '../Box';
import useButtonStyles from './useButtonStyles';
import Spinner from '../Spinner';
import Icon, { IconProps } from '../Icon';

export interface ButtonProps extends NativeAttributes<'button'>, Pick<BoxProps, 'as' | 'to'> {
  /** The size (height) of the button */
  size?: 'small' | 'medium' | 'large';

  /** The variant of the button that decides the colors */
  variant?: 'solid' | 'outline';

  /** The color scheme of the button for solid variants */
  variantColor?:
    | 'blue'
    | 'violet'
    | 'teal'
    | 'green'
    | 'orange'
    | 'red'
    | 'lightred'
    | 'gray'
    | 'darkgray'
    | 'darkblue'
    | 'navyblue'
    | 'lightblue';

  /** The icon present on the button  */
  icon?: IconProps['type'];

  /** The side of the button in which the icon appears on. Defaults to `left`. */
  iconAlignment?: 'left' | 'right';

  /** Whether the button should occupy entirety of its parent. Defaults to `false` */
  fullWidth?: boolean;

  /** Whether the button is disabled */
  disabled?: boolean;

  /** Whether the button should have a loading spinner next to it */
  loading?: boolean;

  /** Whether the button should always be marked as active. Helpful for menus. */
  active?: boolean;
}

/**
 * Extends <a href="/#/Box">Box</a>
 *
 * The core re-usable button that you will use in the app.
 */
export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(function Button(
  {
    size = 'large',
    variant = 'solid',
    variantColor = 'blue',
    loading = false,
    fullWidth = false,
    iconAlignment = 'left',
    active,
    icon,
    children,
    ...rest
  },
  ref
) {
  const styles = useButtonStyles({ variantColor, variant, size });
  const iconSize = size === 'large' ? 'medium' : 'small';

  return (
    <AbstractButton
      aria-pressed={active !== undefined ? active : undefined}
      data-active={active || undefined}
      width={fullWidth ? '100%' : 'auto'}
      {...styles}
      {...rest}
      ref={ref}
    >
      {icon && iconAlignment === 'left' && !loading && (
        <Icon size={iconSize} ml={-1} mr={2} type={icon} />
      )}
      {loading && <Spinner size="small" mr={2} />}
      {children}
      {icon && iconAlignment === 'right' && !loading && (
        <Icon size={iconSize} mr={-1} ml={3} type={icon} />
      )}
    </AbstractButton>
  );
});

export default Button;
