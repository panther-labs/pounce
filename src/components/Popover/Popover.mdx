Popover is a non-modal dialog that floats around a trigger. It's used to display contextual information to the user, and should be paired with a clickable trigger element.

A popover is comprised of 3 elements:

- A `Popoup`
- A `PopupTrigger`
- A `PopupContent`

and are all required items for the popup to appear.

A popover can be simple:

```tsx
<Popover>
  <PopoverTrigger>Click Me</PopoverTrigger>
  <PopoverContent>
    <Card p={6}>Hello</Card>
  </PopoverContent>
</Popover>
```

A popover can have a custom trigger:

```tsx
<Popover>
  <PopoverTrigger as={Button}>Click Me</PopoverTrigger>
  <PopoverContent>
    <Card p={6}>Hello</Card>
  </PopoverContent>
</Popover>
```

A popover can have custom content:

```tsx
const Example = () => {
  const [selectedItem, updateSelectedItem] = React.useState(null);
  return (
    <Popover>
      <PopoverTrigger as={Button}>Click Me</PopoverTrigger>
      <PopoverContent alignment="left-center">
        <Card p={6} width={300}>
          <Combobox
            label="Choose a car manufacturer"
            items={['Toyota', 'Ford', 'Chevrolet', 'BMW', 'Mercedes', 'Hammer', 'Dodge', 'Audi']}
            onChange={updateSelectedItem}
            value={selectedItem}
            placeholder="Select a manufacturer"
          />
        </Card>
      </PopoverContent>
    </Popover>
  );
};

<Example />;
```

A popover can have an alignment:

```tsx
<Popover>
  <PopoverTrigger as={Button}>Click Me</PopoverTrigger>
  <PopoverContent alignment="bottom-center">
    <Card p={6}>Hello from Pounce JS</Card>
  </PopoverContent>
</Popover>
```

A popover can have a spacing from its trigger through the use of `margin`:

```tsx
<Popover>
  <PopoverTrigger as={Button}>Click Me</PopoverTrigger>
  <PopoverContent alignment="right-center" offset={[60, 0]}>
    <Card p={6} mx={5}>
      First Hello from Pounce JS
      <br />
      Second Hello from Pounce JS
      <br />
      Third Hello from Pounce JS
    </Card>
  </PopoverContent>
</Popover>
```

A popover expose its internals for customized use cases:

```tsx
<Popover>
  {({ isOpen, close }) => (
    <React.Fragment>
      <PopoverTrigger as={Button} variantColor={isOpen ? 'violet-400' : 'teal-500'}>
        Click Me
      </PopoverTrigger>
      <PopoverContent alignment="bottom-center">
        <Card p={6}>
          <Text fontSize="medium" mb={2}>
            Hello from Pounce JS
          </Text>
          <Button onClick={close} size="medium">
            Click to close
          </Button>
        </Card>
      </PopoverContent>
    </React.Fragment>
  )}
</Popover>
```

A popover be visible by default:

```tsx
<Popover isOpen>
  <PopoverTrigger as={Button} variantColor="pink-700">
    Click Me
  </PopoverTrigger>
  <PopoverContent alignment="bottom-center">
    <Card p={6}>Hello from Pounce JS</Card>
  </PopoverContent>
</Popover>
```
