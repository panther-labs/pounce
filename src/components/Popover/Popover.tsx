import React from 'react';
import useDisclosure from 'utils/useDisclosure';
import { useId } from '@reach/auto-id';

interface PopoverContext {
  popoverId?: string;
  isOpen: boolean;
  toggle: () => void;
  open: () => void;
  close: () => void;
  triggerRef: React.RefObject<HTMLElement>;
  popoverRef: React.RefObject<HTMLElement>;
}

export interface PopoverProps {
  /**
   * Whether the popover should be initially open
   */
  isOpen?: boolean;

  /**
   * The id of the popover. Helps setting up aria controls between popover content & trigger
   */
  id?: string;

  /**
   * @ignore
   */
  children:
    | React.ReactNode
    | ((
        renderProps: Pick<PopoverContext, 'isOpen' | 'open' | 'close' | 'toggle'>
      ) => React.ReactNode);
}

// @ts-ignore
const PopoverContext = React.createContext<PopoverContext>(undefined);
const usePopoverContext = () => React.useContext(PopoverContext);

const Popover: React.FC<PopoverProps> = ({ isOpen: isInitiallyOpen, id, children }) => {
  const { isOpen, close, open, toggle } = useDisclosure({ isOpen: isInitiallyOpen });
  const triggerRef = React.useRef(null);
  const popoverRef = React.useRef(null);
  const popoverId = useId(id);
  const contextValue = React.useMemo(
    () => ({
      popoverId,
      triggerRef,
      popoverRef,
      open,
      close,
      toggle,
      isOpen,
    }),
    [popoverId, triggerRef, popoverRef, close, toggle, isOpen]
  );

  return (
    <PopoverContext.Provider value={contextValue}>
      {/* @ts-ignore */}
      {typeof children === 'function' ? children({ isOpen, open, close, toggle }) : children}
    </PopoverContext.Provider>
  );
};

export { usePopoverContext };
export default Popover;
