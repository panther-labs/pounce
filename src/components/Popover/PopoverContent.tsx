import React, { ReactNode } from 'react';
import ReachUIPopover from '@reach/popover';
import { useComposedRefs } from '@reach/utils';
import { animated, useTransition } from 'react-spring';
import { usePopoverContext } from './Popover';
import usePopoverContentAlignment from './usePopoverContentAlignment';
import useOutsideClick from '../../utils/useOutsideClick';
import useEscapeKey from '../../utils/useEscapeKey';
import Box from '../Box';

const AnimatedPopover = animated(ReachUIPopover);

export interface PopoverContentProps {
  /**
   * The position to expand the popover
   *
   * @default left
   */
  alignment?:
    | 'bottom-left'
    | 'bottom-center'
    | 'bottom-right'
    | 'top-left'
    | 'top-center'
    | 'top-right'
    | 'left-bottom'
    | 'left-center'
    | 'left-top'
    | 'right-bottom'
    | 'right-center'
    | 'right-top';

  /**
   * Whether the popover should hide when a click outside it occurs
   */
  closeOnOutsideClicks?: boolean;

  /**
   * @ignore
   */
  children: ReactNode;
}

const PopoverContent = React.forwardRef<HTMLDivElement, PopoverContentProps>(
  function PopoverContent(
    { alignment = 'bottom-left', children, closeOnOutsideClicks = true, ...rest },
    forwardedRef
  ) {
    const popoverAlignment = usePopoverContentAlignment(alignment);
    const { popoverId, isOpen, triggerRef, popoverRef, close } = usePopoverContext();

    // When the popoover opens, it should immediately get focus
    React.useEffect(() => {
      window.requestAnimationFrame(() => {
        if (isOpen && popoverRef.current) {
          popoverRef.current.focus();
        }
      });
    }, [isOpen, popoverRef]);

    // Close popover on clicks outside
    useOutsideClick({
      refs: [popoverRef, triggerRef],
      callback: () => isOpen && close(),
      disabled: !closeOnOutsideClicks || !isOpen,
    });

    // Close on ESC key presses
    useEscapeKey({ ref: popoverRef, callback: close, disabled: !isOpen });

    // merge internal + passed prop together
    const ref = useComposedRefs(popoverRef, forwardedRef);

    const transitions = useTransition(isOpen, null, {
      from: { transform: 'translate3d(0, 10px, 0)', opacity: 0, pointerEvents: 'none' },
      enter: { transform: 'translate3d(0, 0, 0)', opacity: 1, pointerEvents: 'auto' },
      leave: { transform: 'translate3d(0, 10px, 0)', opacity: 0, pointerEvents: 'none' },
      config: { duration: 250 },
    });

    return (
      <React.Fragment>
        {transitions.map(
          ({ item, key, props: styles }) =>
            item && (
              <AnimatedPopover
                key={key}
                style={styles}
                targetRef={triggerRef}
                position={popoverAlignment}
                role="tooltip"
                id={popoverId}
                as={'div'}
              >
                <Box ref={ref} tabIndex={0} outline="none" {...rest}>
                  {children}
                </Box>
              </AnimatedPopover>
            )
        )}
      </React.Fragment>
    );
  }
);

export default React.memo(PopoverContent);
