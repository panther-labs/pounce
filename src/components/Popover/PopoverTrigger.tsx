import React from 'react';
import { useComposedRefs, composeEventHandlers } from '@reach/utils';
import type * as Polymorphic from '@reach/utils/polymorphic';
import { NativeAttributes } from '../../system';
import { usePopoverContext } from './Popover';

export type PopoverTriggerProps = NativeAttributes<'button'>;

const PopoverTrigger = React.forwardRef<
  HTMLElement,
  PopoverTriggerProps & { as: React.ElementType }
>(function PopoverTrigger({ as: Comp = 'button', onMouseDown, onKeyDown, ...rest }, forwardedRef) {
  // get the ref that we should store this trigger under
  const { popoverId, triggerRef, toggle } = usePopoverContext();

  // merge internal + passed prop together
  const ref = useComposedRefs(triggerRef, forwardedRef);

  return (
    <Comp
      ref={ref}
      onMouseDown={composeEventHandlers(onMouseDown, toggle)}
      onKeyDown={composeEventHandlers(
        onKeyDown,
        ({ key }: { key: string }) => key === 'Enter' && toggle()
      )}
      aria-describedby={popoverId}
      {...rest}
    />
  );
}) as Polymorphic.ForwardRefComponent<'button', PopoverTriggerProps & { as: React.ElementType }>;

export default React.memo(PopoverTrigger);
