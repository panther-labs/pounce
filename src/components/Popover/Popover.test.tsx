import React from 'react';
import {
  fireClickAndMouseEvents,
  fireEvent,
  renderWithTheme,
  waitForElementToBeRemoved,
  waitMs,
} from 'test-utils';
import Popover from './Popover';
import PopoverTrigger from './PopoverTrigger';
import PopoverMenu from './PopoverContent';
import Button from '../Button';
import Text from '../Text';

describe('Popover', () => {
  it('matches snapshot when closed', () => {
    const { container } = renderWithTheme(
      <Popover id="test">
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when opened', () => {
    const { container } = renderWithTheme(
      <Popover isOpen id="test">
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with alignment', () => {
    const { container } = renderWithTheme(
      <Popover isOpen id="test">
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu alignment="right-center">Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    expect(container).toMatchSnapshot();
  });

  it('only shows the popover trigger by default', () => {
    const { queryByText } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    expect(queryByText('Click me')).toBeInTheDocument();
    expect(queryByText('Boom! I am the popup')).not.toBeInTheDocument();
  });

  it('can render as initially opened', () => {
    const { getByText } = renderWithTheme(
      <Popover isOpen>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    expect(getByText('Boom! I am the popup')).toBeInTheDocument();
  });

  it('toggles with trigger click', async () => {
    const { getByText } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    fireClickAndMouseEvents(trigger);
    await waitForElementToBeRemoved(popup);

    expect(popup).not.toBeInTheDocument();
  });

  it('closes when ECS is pressed', async () => {
    const { getByText } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    fireEvent.keyDown(popup, { key: 'Escape', code: 'Escape' });
    await waitForElementToBeRemoved(popup);
    expect(popup).not.toBeInTheDocument();
  });

  it('closes when a click occurs anywhere outside the popup if `persistOnOutsideClicks` is `false`', async () => {
    const { getByText, container } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu persistOnOutsideClicks={false}>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    // wait for React to properly assign the refs. This is not an issue in real life since a user
    // can't do it faster than React (I tried)
    await waitMs(10);
    fireEvent.mouseDown(container);

    await waitForElementToBeRemoved(popup);
    expect(popup).not.toBeInTheDocument();
  });

  it('does not close when a click occurs outside the popup if `persistOnOutsideClicks` is `true`', async () => {
    const { getByText, container } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu persistOnOutsideClicks>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    // wait for React to properly assign the refs. This is not an issue in real life since a user
    // can't do it faster than React (I tried)
    await waitMs(10);
    fireEvent.mouseDown(container);
    await waitMs(10);

    expect(popup).toBeInTheDocument();
  });

  it('does NOT close when a click occurs inside the popup', async () => {
    const { getByText } = renderWithTheme(
      <Popover>
        <PopoverTrigger as={Button}>Click me</PopoverTrigger>
        <PopoverMenu>Boom! I am the popup</PopoverMenu>
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    fireClickAndMouseEvents(popup);
    await waitMs(20);
    expect(popup).toBeInTheDocument();
  });

  it('correctly exposes its internals', async () => {
    const { getByText } = renderWithTheme(
      <Popover>
        {({ close }) => (
          <React.Fragment>
            <PopoverTrigger as={Button}>Click me</PopoverTrigger>
            <PopoverMenu>
              <Text>Boom! I am the popup</Text>
              <Button onClick={close}>Close popup</Button>
            </PopoverMenu>
          </React.Fragment>
        )}
      </Popover>
    );

    const trigger = getByText('Click me');

    fireClickAndMouseEvents(trigger);
    const popup = getByText('Boom! I am the popup');

    fireClickAndMouseEvents(getByText('Close popup'));
    await waitForElementToBeRemoved(popup);
    expect(popup).not.toBeInTheDocument();
  });
});
