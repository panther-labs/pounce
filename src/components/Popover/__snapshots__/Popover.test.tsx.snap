// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover matches snapshot when closed 1`] = `
.pounce-0 {
  width: auto;
  outline: none;
  padding-left: 16px;
  padding-right: 16px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-0:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-0:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-0:active,
.pounce-0[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

<div>
  <button
    aria-describedby="test"
    class="pounce-0"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Popover matches snapshot when opened 1`] = `
.pounce-0 {
  width: auto;
  outline: none;
  padding-left: 16px;
  padding-right: 16px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-0:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-0:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-0:active,
.pounce-0[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

<div>
  <button
    aria-describedby="test"
    class="pounce-0"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Popover matches snapshot with alignment 1`] = `
.pounce-0 {
  width: auto;
  outline: none;
  padding-left: 16px;
  padding-right: 16px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #0081c5;
  background-color: #0081c5;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.pounce-0:hover {
  background-color: #0a8bcf;
  border-color: #0a8bcf;
}

.pounce-0:focus {
  background-color: #0a8bcf;
  border-color: #55d6ff;
}

.pounce-0:active,
.pounce-0[data-active=true] {
  background-color: #0077bb;
  border-color: #0077bb;
}

<div>
  <button
    aria-describedby="test"
    class="pounce-0"
    type="button"
  >
    Click me
  </button>
</div>
`;
